File: VisualBasicAssignment/screen/sai_keat/widget/order/MyOder.cs
using System;
using System.Data;
using System.Linq;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace VisualBasicAssignment
{
    public partial class MyOder
    {
        public MyOder()
        {
            InitializeComponent();
            _lblDate.Name = "lblDate";
            _dgvOrder.Name = "dgvOrder";
        }

        private void Page_Load(object sender, EventArgs e)
        {
            var u = DatabaseConnections.GetLoginUser();
            using (var db = new FoodShopEntities1())
            {
                try
                {
                    dgvOrder.DataSource = (from d in db.Payments
                                           where d.user_id == u.user_id
                                           orderby d.payment_date descending
                                           select new { d.id, d.total_pay, d.PaymentStatu.name }).ToList();
                }
                catch (Exception ex)
                {
                    Console.WriteLine("Exception caught: {0}", ex);
                }

                try
                {
                    if (dgvOrder.FirstDisplayedCell is object)
                    {
                        var data = Guid.Parse(Conversions.ToString(dgvOrder.FirstDisplayedCell.Value));
                        dgvOrderDetails.DataSource = (from d in db.PaymentDetails
                                                      where d.payment_id == data
                                                      orderby d.food_id ascending
                                                      select new { d.food_id, d.Food.title, d.quantity, d.Food.price }).ToList();
                    }
                }
                catch (Exception ex)
                {
                    Console.WriteLine("Exception caught: {0}", ex);
                }
            }
        }

        private void dgvOrder_CellContentClick(object sender, DataGridViewCellEventArgs e)
        {
            using (var db = new FoodShopEntities1())
            {
                Guid pid = (Guid)dgvOrder.SelectedRows[0].Cells[0].Value;
                var p = db.Payments.Find((object)pid);
                try
                {
                    lblPay.Text = "#" + p.id.ToString();
                    lblDate.Text = "Order Date:" + p.payment_date.ToString();
                    lblStatus.Text = "Status:" + p.PaymentStatu.name;
                    dgvOrderDetails.DataSource = (from d in db.PaymentDetails
                                                  where d.payment_id == pid
                                                  orderby d.food_id ascending
                                                  select new { d.food_id, d.Food.title, d.quantity, d.Food.price }).ToList();
                }
                catch (Exception ex)
                {
                    Console.WriteLine("Exception caught: {0}", ex);
                }
            }
        }

        private void Guna2Button1_Click(object sender, EventArgs e)
        {
        }

        private void lblDate_Click(object sender, EventArgs e)
        {
        }
    }
}

File: VisualBasicAssignment/screen/sai_keat/widget/order/MyOder.vb
﻿Public Class MyOder

    Private Sub Page_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim u As User = DatabaseConnections.GetLoginUser()
        Using db As FoodShopEntities1 = New FoodShopEntities1()
            Try
                Me.dgvOrder.DataSource = (
                                      From d In db.Payments
                                      Where d.user_id = u.user_id
                                      Order By d.payment_date Descending
                                      Select New With {d.id, d.total_pay, d.payment_status}
                                      ).ToList()

            Catch ex As Exception
                Console.WriteLine("Exception caught: {0}", ex)
            End Try
            Try
                If dgvOrder.FirstDisplayedCell IsNot Nothing Then
                    Dim data As Guid = Guid.Parse(dgvOrder.FirstDisplayedCell.Value)
                    Me.dgvOrderDetails.DataSource = (From d In db.PaymentDetails
                                                     Where d.payment_id = data
                                                     Order By d.food_id Ascending
                                                     Select New With {d.food_id, d.Food.title, d.quantity, d.Food.price}).ToList()
                End If
            Catch ex As Exception
                Console.WriteLine("Exception caught: {0}", ex)
            End Try
        End Using
    End Sub

    Private Sub dgvOrder_CellContentClick(sender As Object, e As DataGridViewCellEventArgs) Handles dgvOrder.CellClick

        Using db As FoodShopEntities1 = New FoodShopEntities1()
            Dim pid As Guid = dgvOrder.SelectedRows.Item(0).Cells.Item(0).Value
            Try
                Me.dgvOrderDetails.DataSource = (From d In db.PaymentDetails
                                                 Where d.payment_id = pid
                                                 Order By d.food_id Ascending
                                                 Select New With {d.food_id, d.Food.title, d.quantity, d.Food.price}).ToList()
            Catch ex As Exception
                Console.WriteLine("Exception caught: {0}", ex)
            End Try
        End Using
    End Sub

    Private Sub Guna2Button1_Click(sender As Object, e As EventArgs) Handles Guna2Button1.Click

    End Sub
End Class
